### PR TITLE
[FLINK-23423][docs-zh] Translate the page of "Elasticsearch Connector" into Chinese

### PR DESCRIPTION
**Pull request by huxixiang**
## What is the purpose of the change
Translate the page of "Elasticsearch Connector" into Chinese

The page url is "https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/connectors/datastream/elasticsearch/".

## Brief change log

Translate "flink/docs/content.zh/docs/connectors/datastream/elasticsearch.md"

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (no)

### CODE REVIEW

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 1]
```diff
@@ -396,13 +374,13 @@ input.addSink(new ElasticsearchSink(
                 RequestIndexer indexer) {
 
             if (ExceptionUtils.findThrowable(failure, EsRejectedExecutionException.class).isPresent()) {
-                // full queue; re-add document for indexing
+                 //队列已满；重新添加文档进行索引
```
Comment (movesan) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:377]: 注释缩进问题

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 2]
```diff
@@ -396,13 +374,13 @@ input.addSink(new ElasticsearchSink(
...
                 indexer.add(action)
             } else if (ExceptionUtils.findThrowable(failure, ElasticsearchParseException.class).isPresent()) {
-                // malformed document; simply drop request without failing sink
+                 //文档格式错误；简单地删除请求避免接收器失败
```
Comment (movesan) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:380]: 同上

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 3]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
- 
- * **bulk.flush.backoff.enable**: Whether or not to perform retries with backoff delay for a flush
- if one or more of its actions failed due to a temporary `EsRejectedExecutionException`.
- * **bulk.flush.backoff.type**: The type of backoff delay, either `CONSTANT` or `EXPONENTIAL`
- * **bulk.flush.backoff.delay**: The amount of delay for backoff. For constant backoff, this
- is simply the delay between each retry. For exponential backoff, this is the initial base delay.
- * **bulk.flush.backoff.retries**: The amount of backoff retries to attempt.
+ * **bulk.flush.max.actions**：刷新前缓存的最大操作数。
+ * **bulk.flush.max.size.mb**：刷新前缓存的最大数据大小（以兆字节为单位）。
+ * **bulk.flush.interval.ms**：不论缓存操作的数量或大小如何，刷新的时间间隔。
```
Comment (movesan) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:414]: "不论缓存操作的数量或大小如何，刷新的时间间隔。" 是否可进行语序调整，如："刷新的时间间隔（不论缓存操作的数量或大小如何）。"

**Review by movesan — COMMENTED** [review]
注释字符 "//" 与字符之间应存在空格

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md (hunk 3 above)
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:414]: 感谢，我修改一下。

**Comment by huxixiang** [comment]
cc @wuchong , could you review my PR when you're free?  Looking for your feedbacks, thanks a lot.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 4]
```diff
@@ -276,49 +270,37 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
-### Elasticsearch Sinks and Fault Tolerance
+### Elasticsearch 接收器和容错
 
-With Flink’s checkpointing enabled, the Flink Elasticsearch Sink guarantees
-at-least-once delivery of action requests to Elasticsearch clusters. It does
-so by waiting for all pending action requests in the `BulkProcessor` at the
-time of checkpoints. This effectively assures that all requests before the
-checkpoint was triggered have been successfully acknowledged by Elasticsearch, before
-proceeding to process more records sent to the sink.
+启用 Flink 的检查点后，Flink Elasticsearch 接收器保证至少一次将操作请求发送到 Elasticsearch 集群。
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:297]: 保持“checkpointing”这个词，无需翻译为“检查点”，参见 [Checkpointing 中文文档](https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/datastream/fault-tolerance/checkpointing/)，下文中的 checkpoint 也是如此。

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 5]
```diff
@@ -276,49 +270,37 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
+它通过在检查点时等待 `BulkProcessor` 中所有挂起的操作请求来实现。
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:298]: “它” -> “该特性”？

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 6]
```diff
@@ -276,49 +270,37 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
+这有效地保证了在触发检查点之前所有的请求被 Elasticsearch 成功确认，然后继续处理发送到接收器的更多的记录。
 
-More details on checkpoints and fault tolerance are in the [fault tolerance docs]({{< ref "docs/learn-flink/fault_tolerance" >}}).
+检查点和容错更多详细信息，请参见[容错文档]({{< ref "docs/learn-flink/fault_tolerance" >}})。
 
-To use fault tolerant Elasticsearch Sinks, checkpointing of the topology needs to be enabled at the execution environment:
+要使用容错 Elasticsearch Sinks，需要在执行环境启用拓扑检查点：
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:303]: 这里的“topology”应该指的是 Flink 的作业拓扑图，“拓扑检查点”容易让用户误解为一种特别的 checkpoint。结合上述评论，可以考虑一下如何修改。

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 7]
```diff
@@ -336,21 +318,17 @@ env.enableCheckpointing(5000) // checkpoint every 5000 msecs
 {{< /tabs >}}
 
 <p style="border-radius: 5px; padding: 5px" class="bg-danger">
-<b>NOTE</b>: Users can disable flushing if they wish to do so, by calling
-<b>disableFlushOnCheckpoint()</b> on the created <b>ElasticsearchSink</b>. Be aware
-that this essentially means the sink will not provide any strong
-delivery guarantees anymore, even with checkpoint for the topology enabled.
+<b>注意</b>： 如果用户愿意，可以通过在创建的
+<b> ElasticsearchSink </b>上调用 <b>disableFlushOnCheckpoint()</b> 来禁用刷新。请注意，
+ 这实质上意味着接收器将不再提供任何强大的交付保证，即使启用了拓扑检查点。
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:323]: “strong delivery guarantees”指的是保证 exactly-once 地向 es 写入数据，“强大的交付保证”让人不太能理解...

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 8]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
 {{< /tab >}}
 {{< /tabs >}}
 
-The above example will let the sink re-add requests that failed due to
-queue capacity saturation and drop requests with malformed documents, without
-failing the sink. For all other failures, the sink will fail. If a `ActionRequestFailureHandler`
-is not provided to the constructor, the sink will fail for any kind of error.
+上面的示例接收器重新添加由于队列容量饱和而失败的请求并丢弃文档格式错误的请求，而不会使接收器失败。
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:392]: -> “上述的示例接收器会重新添加由于队列容量饱和而失败的请求，同时丢弃文档格式错误的请求，而...”

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 9]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
+对于其它故障，接收器将会失败。如果未向构造器提供一个 `ActionRequestFailureHandler`，那么任何类型的错误都会导致接收器失败。
 
-Note that `onFailure` is called for failures that still occur only after the
-`BulkProcessor` internally finishes all backoff retry attempts.
-By default, the `BulkProcessor` retries to a maximum of 8 attempts with
-an exponential backoff. For more information on the behaviour of the
-internal `BulkProcessor` and how to configure it, please see the following section.
+注意，`onFailure` 仅在 `BulkProcessor` 内部完成所有补偿重试尝试后仍发生故障时被调用。
+默认情况下，`BulkProcessor` 最多重试 8 次，并采用指数补偿。有关 `BulkProcessor` 内部行为以及如何配置它的更多信息，请参阅以下部分。
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:396]: “并采用指数补偿” -> “两次重试之间的等待时间呈指数级增长”。下文出现的“补偿”也应重新考虑翻译方式。

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 10]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
-For the execution of your Flink program, it is recommended to build a
-so-called uber-jar (executable jar) containing all your dependencies
-(see [here]({{< ref "docs/dev/datastream/project-configuration" >}}) for further information).
+可以在[此处](https://elastic.co)找到 Elasticsearch 的更多信息。
 
-Alternatively, you can put the connector's jar file into Flink's `lib/` folder to make it available
-system-wide, i.e. for all job being run.
+## 将 Elasticsearch 连接器打包到 Uber-Jar 中
+
+为了执行你的 Flink 程序，建议构建一个叫做 uber-jar (可执行的 jar)，其中包含了你所有的依赖
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:427]: 去掉“叫做”

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 11]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
+(更多信息参见[此处]({{< ref "docs/dev/datastream/project-configuration" >}}))。
+
+或者，你可以将连接器的 jar 文件放入 Flink 的 `lib/` 目录下，使其在系统范围内可用，即用于所有正在运行的作业。
```
Comment (tsreaper) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:430]: “系统范围‘ -> “全局范围”。

“being run”不是进行时，是表修饰的被动语态。“即用于所有正在运行的作业” -> “即用于所有作业”。

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md (hunk 7 above)
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:323]: 感谢，这里强大的交付保证是指上文的at-least-once

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md (hunk 11 above)
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md]: 感谢，已根据您的反馈进行了修改。

**Comment by huxixiang** [comment]
cc @RocMarshal, could you review my pr when you're free? Looking for your feedbacks, thanks a lot.

**Comment by RocMarshal** [comment]
Thank you for your contribution.I'll check this pr, wait a moment @huxixiang

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 12]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
-For the execution of your Flink program, it is recommended to build a
-so-called uber-jar (executable jar) containing all your dependencies
-(see [here]({{< ref "docs/dev/datastream/project-configuration" >}}) for further information).
+可以在[此处](https://elastic.co)找到 Elasticsearch 的更多信息。
 
-Alternatively, you can put the connector's jar file into Flink's `lib/` folder to make it available
-system-wide, i.e. for all job being run.
+## 将 Elasticsearch 连接器打包到 Uber-Jar 中
+
+为了执行你的 Flink 程序，建议构建一个 uber-jar (可执行的 jar)，其中包含了你所有的依赖
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:427]:
```suggestion
建议构建一个包含程序所有依赖的 uber-jar (可执行的 jar)，以便更好地执行你的 Flink 程序。
```
Only a minor suggestion.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 13]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
+(更多信息参见[此处]({{< ref "docs/dev/datastream/project-configuration" >}}))。
+
+或者，你可以将连接器的 jar 文件放入 Flink 的 `lib/` 目录下，使其在全局范围内可用，即用于所有的作业。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/datastream/elasticsearch.md:430]:
```suggestion
或者，你可以将连接器的 jar 文件放入 Flink 的 `lib/` 目录下，使其在全局范围内可用，即可用于所有的作业。
```

**Review by RocMarshal — COMMENTED** [review]
@huxixiang Hi, so sorry for my late dealing. And I left some comments. Please let me know what's your opinion.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 14]
```diff
@@ -51,29 +49,25 @@ of the Elasticsearch installation:
...
 
-## Installing Elasticsearch
+## 安装 Elasticsearch
 
-Instructions for setting up an Elasticsearch cluster can be found
-[here](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html).
-Make sure to set and remember a cluster name. This must be set when
-creating an `ElasticsearchSink` for requesting document actions against your cluster.
+Elasticsearch 集群的设置可以参考[此处](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html)。
+确保设置并记住集群名称。这是在创建 `ElasticsearchSink` 请求集群文档操作时必须要设置的。
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:64]: Accepted, Thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 15]
```diff
@@ -51,29 +49,25 @@ of the Elasticsearch installation:
...
 
 ## Elasticsearch Sink
 
-The `ElasticsearchSink` uses a `TransportClient` (before 6.x) or `RestHighLevelClient` (starting with 6.x) to communicate with an
-Elasticsearch cluster.
+`ElasticsearchSink` 使用 `TransportClient` （6.x 之前） 或者 `RestHighLevelClient` （6.x 开始） 和 Elasticsearch 集群进行通信。
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:68]: Accepted, Thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 16]
```diff
@@ -98,7 +92,7 @@ DataStream<String> input = ...;
 
 Map<String, String> config = new HashMap<>();
 config.put("cluster.name", "my-cluster-name");
-// This instructs the sink to emit after every element, otherwise they would be buffered
+// 这指示 sink 在接收每个元素之后立即提交，否则它们将被缓存
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:95]: Accepted, I have modified the previous translation to
// 下面的设置使 sink 在接收每个元素之后立即提交，否则这些元素将被缓存起来

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 17]
```diff
@@ -166,10 +160,10 @@ ElasticsearchSink.Builder<String> esSinkBuilder = new ElasticsearchSink.Builder<
     }
 );
 
-// configuration for the bulk requests; this instructs the sink to emit after every element, otherwise they would be buffered
+// 批量请求的配置；这指示 sink 在接收每个元素之后立即提交，否则它们将被缓存
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:163]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 18]
```diff
@@ -276,81 +270,65 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
-For Elasticsearch versions that still uses the now deprecated `TransportClient` to communicate
-with the Elasticsearch cluster (i.e., versions equal or below 5.x), note how a `Map` of `String`s
-is used to configure the `ElasticsearchSink`. This config map will be directly
-forwarded when creating the internally used `TransportClient`.
-The configuration keys are documented in the Elasticsearch documentation
-[here](https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html).
-Especially important is the `cluster.name` parameter that must correspond to
-the name of your cluster.
+对于仍然使用已被弃用的 `TransportClient` 和 Elasticsearch 集群通信的 Elasticsearch 版本 (即，小于或等于 5.x 的版本)，
+请注意如何使用一个 `String` 类型的 `Map` 配置 `ElasticsearchSink`。在创建内部使用的 `TransportClient` 时将直接转发此配置映射。
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:280]: Accepted, 
I have modified the previous translation to
对于仍然使用已被弃用的 `TransportClient` 和 Elasticsearch 集群通信的 Elasticsearch 客户端版本 (即，小于或等于 5.x 的版本)，
使用一个 `String` 类型的 `Map` 配置 `ElasticsearchSink`。这些配置项将在创建 `TransportClient` 时被使用。

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 19]
```diff
@@ -276,81 +270,65 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
+配置项参见[此处](https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html)的 Elasticsearch 文档中。
+特别重要的是参数 `cluster.name` 必须和你的集群名称对应上。
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:282]: Accepted, thanks

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 20]
```diff
@@ -276,81 +270,65 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
+另外注意，该示例仅演示了对每个传入的元素执行单个索引请求。
+通常，`ElasticsearchSinkFunction` 可用于执行多个不同类型的请求（例如 `DeleteRequest`、 `UpdateRequest` 等）。
 
-Internally, each parallel instance of the Flink Elasticsearch Sink uses
-a `BulkProcessor` to send action requests to the cluster.
-This will buffer elements before sending them in bulk to the cluster. The `BulkProcessor`
-executes bulk requests one at a time, i.e. there will be no two concurrent
-flushes of the buffered actions in progress.
+在内部，Flink Elasticsearch Sink 的每个并行实例使用一个 `BulkProcessor` 向集群发送操作请求。
+这将使得元素在发送到集群之前进行批量缓存。
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:292]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 21]
```diff
@@ -276,81 +270,65 @@ esSinkBuilder.setRestClientFactory(new RestClientFactory {
...
 ```
 {{< /tab >}}
 {{< /tabs >}}
 
 <p style="border-radius: 5px; padding: 5px" class="bg-danger">
-<b>NOTE</b>: Users can disable flushing if they wish to do so, by calling
-<b>disableFlushOnCheckpoint()</b> on the created <b>ElasticsearchSink</b>. Be aware
-that this essentially means the sink will not provide any strong
-delivery guarantees anymore, even with checkpoint for the topology enabled.
+<b>注意</b>： 如果用户愿意，可以通过在创建的
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:321]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 22]
```diff
@@ -396,13 +374,13 @@ input.addSink(new ElasticsearchSink(
                 RequestIndexer indexer) {
 
             if (ExceptionUtils.findThrowable(failure, EsRejectedExecutionException.class).isPresent()) {
-                // full queue; re-add document for indexing
+                 // 队列已满；重新添加文档进行索引
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:377]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 23]
```diff
@@ -396,13 +374,13 @@ input.addSink(new ElasticsearchSink(
...
                 indexer.add(action)
             } else if (ExceptionUtils.findThrowable(failure, ElasticsearchParseException.class).isPresent()) {
-                // malformed document; simply drop request without failing sink
+                 // 文档格式错误；简单地删除请求避免 sink 失败
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:380]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md [hunk 24]
```diff
@@ -411,61 +389,44 @@ input.addSink(new ElasticsearchSink(
...
 
 <p style="border-radius: 5px; padding: 5px" class="bg-danger">
-<b>IMPORTANT</b>: Re-adding requests back to the internal <b>BulkProcessor</b>
-on failures will lead to longer checkpoints, as the sink will also
-need to wait for the re-added requests to be flushed when checkpointing.
-For example, when using <b>RetryRejectedExecutionFailureHandler</b>, checkpoints
-will need to wait until Elasticsearch node queues have enough capacity for
-all the pending requests. This also means that if re-added requests never
-succeed, the checkpoint will never finish.
+<b>重要提示</b>：在失败时将请求重新添加回内部 <b>BulkProcessor</b> 会导致更长的 checkpoint，因为在进行 checkpoint 时， sink 还需要等待重新添加的请求被刷新。
```
[the remark this answers is not in the document]
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:402]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md (hunk 12 above)
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:427]: Accepted, thanks.

File: docs/content.zh/docs/connectors/datastream/elasticsearch.md (hunk 13 above)
Comment (huxixiang) [reply docs/content.zh/docs/connectors/datastream/elasticsearch.md:430]: Accepted, thanks.

**Review by RocMarshal — APPROVED** [review]
@huxixiang Thanks for update. LGTM +1. Now ping @wuchong .

**Comment by wuchong** [comment]
Merging...

**Pull request merged**